### DISCUSSION TITLE
How does pocketbase verify token?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#683] [Q&A]
Does pocketbase verify token? How?

**Comment by ganigeorgiev** [comment]
PocketBase is using JWTs. The verification process differ depending on what is the token type (auth, password reset, account verification, email change confirmation, etc.).

If you are looking for how to authenticate, you could find some info in https://pocketbase.io/docs/api-authentication/.

If you are looking for where the token generation and verification happens in the code:
- token factories - https://github.com/pocketbase/pocketbase/blob/master/tokens/user.go
- how we locate a single user by a token - https://github.com/pocketbase/pocketbase/blob/master/daos/user.go#L122-L152
- where the actual parsing and verification happens - https://github.com/pocketbase/pocketbase/blob/master/tools/security/jwt.go
